test/suite/test_cursor_bound11.py

```diff
@@ -0,0 +1,334 @@
+        new_skip_count = self.get_stat(stat.conn.cursor_next_skip_total)
+        self.assertEqual(new_skip_count - skip_count, 0)
+
+    # Test a basic prepared scenario.
+    def test_prepared(self):
+        uri = 'table:test_base_scenario'
+        self.session.create(uri, 'key_format=u,value_format=u')
+        cursor = self.session.open_cursor(uri)
+        session2 = self.conn.open_session()
+        cursor3 = session2.open_cursor(uri, None, "debug=(release_evict=true)")
+        # Insert an update without timestamp
+        l = "abcdefghijklmnopqrstuvwxyz"
+        session2.begin_transaction()
+        key_count = 0
+
+        # Insert 'cc'
+        self.session.begin_transaction()
+        cursor['cc'] = 'cc'
+        self.session.commit_transaction()
+
+        # Prepare keys aa -> zz but not 'cX'
+        self.session.begin_transaction()
+        for i in range (0, 26):
+            if (i == 2):
+                continue
+            for j in range (0, 26):
+                cursor[l[i] + l[j]] = l[i] + l[j]
+                key_count += 1
+        cursor.reset()
+        self.session.prepare_transaction('prepare_timestamp=2')
+
+        # Evict the whole range.
+        for i in range (0, 26):
+            for j in range(0, 26):
+                cursor3.set_key(l[i] + l[j])
+                cursor3.search()
+                cursor3.reset()
+
+        # Search near for the 'c' key. We can't see this as our txn started before 'cc' was inserted.
+        cursor2 = session2.open_cursor(uri)
+        cursor2.set_key('c')
+        self.assertEqual(cursor2.search_near(), wiredtiger.WT_NOTFOUND)
+
+        skip_count = self.get_stat(stat.conn.cursor_next_skip_total, session2)
+        # This should be equal to roughly key_count as we're going to traverse the whole
+        # range forwards. Not including 'a' and 'b'.
+        self.assertGreaterEqual(skip_count, key_count - 2*26)
+
+        set_prefix_bound(self, cursor2, 'c')
+        cursor2.set_key('c')
+        self.assertEqual(cursor2.search_near(), wiredtiger.WT_NOTFOUND)
+
+        bound_skip_count = self.get_stat(stat.conn.cursor_next_skip_total, session2)
+        # We expect to traverse one entry, 'cc'.
+        self.assertEqual(bound_skip_count - skip_count, 1)
+        skip_count = bound_skip_count
+
+        # We early exit here as "cc" is not the last key. 
+        self.assertEqual(self.get_stat(stat.conn.cursor_bounds_next_early_exit, session2), 1)
+
+        session2.rollback_transaction()
+        session2.begin_transaction('ignore_prepare=true')
+        cursor4 = session2.open_cursor(uri)
+        set_prefix_bound(self, cursor4, 'c')
+        cursor4.set_key('c')
+        self.assertEqual(cursor4.search_near(), 1)
+        bound_skip_count = self.get_stat(stat.conn.cursor_next_skip_total, session2)
+        # We expect to not skip any entries and return 'cc'
+        self.assertEqual(bound_skip_count - skip_count, 0)
+        self.assertEqual(cursor4.get_key(), b'cc')
```

Comment:
np: Why not use self.check_key() here?

Reply:
🤔 good point.

Reply:
This test wasn't built with `wtbound` functionality in mind, as such check_key doesn't work here.